stdlib/list.gr

```diff
@@ -262,42 +333,63 @@ let rec take = (n, list) => {
     match ((n, list)) {
       (_, []) => list,
       (n, _) when n == 0 => [],
-      (n, [first, ...rest]) => [first, ...take(n-1, rest)]
+      (n, [first, ...rest]) => [first, ...take(n - 1, rest)],
     }
   }
 }
 
 let rec takeWhile = (p, list) => {
   match (list) {
     [] => [],
-    [first, ...rest] => if (p(first)) [first, ...takeWhile(p, rest)] else []
+    [first, ...rest] =>
+      if (p(first)) {
+        [first, ...takeWhile(p, rest)]
+      } else {
+        []
+      },
   }
 }
 
 let rec find = (fn, list) => {
   match (list) {
     [] => None,
-    [first, ...rest] => if (fn(first)) Some(first) else find(fn, rest)
+    [first, ...rest] =>
+      if (fn(first)) {
+        Some(first)
+      } else {
+        find(fn, rest)
+      },
   }
 }
 
 let findIndex = (fn, list) => {
   let rec findItemIndex = (l, index) => {
     match (l) {
       [] => None,
-      [first, ...rest] => if (fn(first)) Some(index) else findItemIndex(rest, index + 1)
+      [first, ...rest] =>
+        if (fn(first)) {
+          Some(index)
+        } else {
+          findItemIndex(rest, index + 1)
+        },
     }
   }
   findItemIndex(list, 0)
 }
 
 let product = (a, b) => {
   let mut list = []
-  forEach((aItem) => {
-    forEach((bItem) => {
-     list = [(aItem, bItem), ...list]
-    },b)
-  }, a)
+  forEach(
+    aItem => {
+      forEach(
+        bItem => {
+          list = [(aItem, bItem), ...list]
+        },
+        b,
+      )
+    },
+    a,
+  )
```

Comment:
This looks like it broke just about everywhere it could, but I don't think it needed to?

Reply:
It doesn't need to but because the block broke it propagates up the parent chain.  I need to apply some grouping to stop this happening.

Reply:
Looks like this is a common formatting approach, here's how it looks for reasonml:

![image](https://user-images.githubusercontent.com/15860044/135709593-f78fc5cb-13ae-4718-8236-1dfc5ec43a1f.png)